subsys/ipc/CMakeLists.txt

```diff
@@ -2,3 +2,5 @@
 
 add_subdirectory_ifdef(CONFIG_RPMSG_SERVICE rpmsg_service)
 add_subdirectory_ifdef(CONFIG_IPC_SERVICE ipc_service)
+
+zephyr_sources_ifdef(CONFIG_ICMSG icmsg.c)
```

Comment:
If we are moving `icmsg` to be a library why do not move that to `subsys/ipc/ipc_service/lib` instead of polluting the `subsys/ipc` root?

Reply:
+1

Reply:
I was thinking that icmsg could be used in places than an ipc_service backend. But I don't have any use-cases in mind so I'll move it to `.../ipc_service/lib`.
If we find such use cases we could discuss where to move it.